File: src/main/java/io/github/dqualizer/dqlang/types/rqa/configuration/loadtest/LoadTestArtifact.java
package io.github.dqualizer.dqlang.types.rqa.configuration.loadtest;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.github.dqualizer.dqlang.types.dam.architecture.RESTEndpoint;
import io.github.dqualizer.dqlang.types.rqa.definition.loadtest.Artifact;
import io.github.dqualizer.dqlang.types.rqa.definition.loadtest.ResponseMeasures;
import io.github.dqualizer.dqlang.types.rqa.definition.stimulus.Stimulus;
import lombok.*;

@Getter
@ToString
@AllArgsConstructor
@NoArgsConstructor
public class LoadTestArtifact {

    private Artifact artifact;
    @NonNull
    private String description;
    private Stimulus stimulus;
    @JsonProperty("response_measure")
    private ResponseMeasures responseMeasure;
    private RESTEndpoint HTTPEndpoint;
}

File: src/main/kotlin/io/github/dqualizer/dqlang/types/rqa/configuration/loadtest/LoadTestArtifact.kt
package io.github.dqualizer.dqlang.types.rqa.configuration.loadtest

import com.fasterxml.jackson.annotation.JsonProperty
import io.github.dqualizer.dqlang.types.dam.architecture.RESTEndpoint
import io.github.dqualizer.dqlang.types.rqa.definition.loadtest.Artifact
import io.github.dqualizer.dqlang.types.rqa.definition.loadtest.ResponseMeasures
import io.github.dqualizer.dqlang.types.rqa.definition.stimulus.Stimulus

data class LoadTestArtifact (
    val artifact: Artifact? = null,

    val description: String? = null,

    val stimulus: Stimulus? = null,

    @JsonProperty("response_measure")
    val responseMeasure: ResponseMeasures? = null,

    val httpEndpoint: RESTEndpoint? = null
)
